File: IntegrationRNObjC/ios/IntegrationObjC/ViewController.m
//
//  ViewController.m
//  IntegrationObjC
//
//  Created by VuNguyenIT on 2/28/18.
//

#import "ViewController.h"
#import <React/RCTRootView.h>

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
//    NSLog(@"navigate RN Button Pressed");
//    NSURL *jsCodeLocation = [NSURL URLWithString:@"http://localhost:8081/index.bundle?platform=ios"];
//    
//    RCTRootView *rootView =
//    [[RCTRootView alloc] initWithBundleURL: jsCodeLocation
//                                moduleName: @"IntegrationRNObjC"
//                         initialProperties:nil
//                             launchOptions: nil];
//    UIViewController *vc = [[UIViewController alloc] init];
//    vc.view = rootView;
//    [self presentViewController:vc animated:YES completion:nil];
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

- (IBAction)navigateToRNPress:(UIButton *)sender {
    NSLog(@"navigate RN Button Pressed");
    
    [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(backToNative) name:@"dismissViewController" object:nil];
    
    NSURL *jsCodeLocation = [NSURL URLWithString:@"http://localhost:8081/index.bundle?platform=ios"];
    
    RCTRootView *rootView =
    [[RCTRootView alloc] initWithBundleURL: jsCodeLocation
                                moduleName: @"IntegrationRNObjC"
                         initialProperties:nil
                             launchOptions: nil];
    UIViewController *vc = [[UIViewController alloc] init];
    vc.view = rootView;
    [self presentViewController:vc animated:YES completion:nil];
}

- (void)backToNative {
    [self dismissViewControllerAnimated:YES completion:nil];
}
@end

File: IntegrationRNSwift/ios/IntegrationSwift/ViewController.swift
//
//  ViewController.swift
//  IntegrationSwift
//
//  Created by VuNguyenIT on 3/1/18.
//  Copyright © 2018 VuNguyenIT. All rights reserved.
//

import UIKit
import React

class ViewController: UIViewController {
    
    var integrationRNSwift: RCTRootView!

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }

    @IBAction func navigateRNPress(_ sender: Any) {
        NotificationCenter.default.addObserver(self, selector: #selector(backToNative), name: Notification.Name("dismissRNViewController"), object: nil)
        
        integrationRNSwift = MixerReactModule.sharedInstance.navigateToRNModule(
            "IntegrationRNSwift",
            initialProperties: nil)
        let rnController = ViewController()
        rnController.view = integrationRNSwift
        self.present(rnController, animated: true, completion: nil)
//        integrationRNSwift.frame = self.view.bounds
//        self.view.addSubview(integrationRNSwift)
    }
    
    @objc func backToNative(){
        self.dismiss(animated: true, completion: nil)
        //self.integrationRNSwift.removeFromSuperview()
    }
    
}

